DOC-11796 Clarify that Prometheus endpoint is not deprecated (#19187)
cockroachdb/docs: src/current/v23.1/monitoring-and-alerting.md

### Raw status endpoints

{{site.data.alerts.callout_info}}
The JSON endpoints are deprecated in favor of the [Cluster API](#cluster-api).

The `/_status/vars` metrics endpoint is in Prometheus format and is not deprecated. For more information, refer to [Prometheus endpoint](#prometheus-endpoint).
{{site.data.alerts.end}}

> Several endpoints return raw status meta information in JSON at `http://<host>:<http-port>/#/debug`. Feel free to investigate and use these endpoints, but note that they are subject to change.

Reviewer: ```suggestion
Several endpoints return raw status meta information in JSON at `http://<host>:<http-port>/#/debug`. You can investigate and use these endpoints, but note that they are subject to change.
```

Not your change here, I know, but "Feel free" feels kind of informal to me.

Reviewer: Same for each version, if you decide to change.

Author: good suggestion. updated.
